File: games/rockpaperscissors.py
import random

numGames = int(input("How many games would you like to play? "))
print()

wins = 0
losses = 0
ties = 0

moveMap = {0: "rock", 1: "paper", 2: "scissors"}

for i in range(numGames):
  print("Round " + str(i+1) + " of " + str(numGames))
  print("-------------------------")
  playerMove = input("Please enter your move (rock, paper or scissors): ")

  while(playerMove != "rock" and playerMove != "paper" and playerMove != "scissors"):
    playerMove = input("Invalid move. Try again: ")

  cpuMove = moveMap[random.randint(0, 2)]
  print("The computer has chosen " + cpuMove + ".", end=" ")

  if(playerMove == cpuMove):
    ties += 1
    print("You have tied this round!")
  elif((playerMove == "rock" and cpuMove == "scissors") or (playerMove == "scissors" and cpuMove == "paper") or (playerMove == "paper" and cpuMove == "rock")):
    wins += 1
    print("You win this round!")
  else:
    losses += 1
    print("You have lost this round!")

  print("Wins: " + str(wins) + ", losses: " + str(losses) + ", ties: " + str(ties))
  print()

File: games/rockpaperscissors.java
import java.util.Scanner;
import java.util.Random;

class Main {
  public static void main(String[] args) {
    Scanner input = new Scanner(System.in);
    Random rand = new Random();
    String[] moves = {"rock", "paper", "scissors"};
    
    System.out.println("How many games do you want to play?: ");
    int numGames = input.nextInt();
    input.nextLine();

    int wins = 0, losses = 0, ties = 0;

    for(int i = 1; i <= numGames; i++){
      System.out.println("Round " + i + " of " + numGames);

      System.out.println("Enter your move (rock, paper or scissors): ");
      String playerMove = input.nextLine();
      while(!playerMove.equals("rock") && !playerMove.equals("paper") && !playerMove.equals("scissors")){
        System.out.println("Invalid move. Please try again: ");
        playerMove = input.nextLine();
      }

      String cpuMove = moves[rand.nextInt(3)];
      System.out.print("The computer chose " + cpuMove + ". ");

      if(cpuMove.equals(playerMove)){
        ties++;
        System.out.println("You tied this round!");
      } else if(
        (cpuMove.equals("rock") && playerMove.equals("paper")) ||
        (cpuMove.equals("paper") && playerMove.equals("scissors")) ||
        (cpuMove.equals("scissors") && playerMove.equals("rock"))
      ){
        wins++;
        System.out.println("You won this round!");
      } else {
        losses++;
        System.out.println("You lost this round!");
      }

      System.out.println("Wins: " + wins + ", losses: " + losses + ", ties: " + ties);
      System.out.println();
    }

    input.close();
  }
}